cpp/src/arrow/memory_pool.h

```diff
@@ -86,6 +86,25 @@ class ARROW_EXPORT LoggingMemoryPool : public MemoryPool {
   MemoryPool* pool_;
 };
 
+class ARROW_EXPORT ProxyMemoryPool : public MemoryPool {
```

Comment:
Can we consider some alternative names than "Proxy"? Can you add a doxygen comment describing what this class is?

Reply:
How about SubMemoryPool? MemorySubPool? PartialMemoryPool?